text/src/main/scala/akka/stream/alpakka/text/impl/CharsetCanonicalizingFlow.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.text.impl
+
+import java.nio.charset.{Charset, StandardCharsets}
+
+import akka.NotUsed
+import akka.annotation.InternalApi
+import akka.stream.scaladsl.{Flow, Source}
+import akka.util.ByteString
+
+@InternalApi
+private[text] class CharsetCanonicalizingFlow(outgoing: Charset) {
+
+  // see: https://en.wikipedia.org/wiki/Byte_order_mark
+  private val ensureByteOrderMark: Flow[ByteString, ByteString, NotUsed] =
+    Flow[ByteString].prefixAndTail(1).flatMapConcat {
```

Comment:
@ennru How do you feel about using the scaladsl flows here? I see the other implementations are implemented in GraphStageLogic and was wondering if we should rewrite this in terms of graphstage logic directly.

Reply:
Use the least powerful tool that does the job. This looks good to me.